### ISSUE TITLE
Duplicate subscriptions towards the same path

### ISSUE THREAD

**Issue by fredsod** [karimra/gnmic#486]
Hi,

When I try to start a subscription, GNMIc creates two subscriptions towards the same router and path, I can't really figure out why.

I'm running GNMIc with the following command:
`gnmic -a <IP> -u <username> sub -e json --tls-cert cert.pem --tls-key key.pem --tls-ca ca-cert.pem --path <path> --stream-mode sample -d`

Debug log when the duplication occurs:
```
2021/11/01 21:05:44.340551 /build/config/subscriptions.go:44: [config] subscriptions: map[default-1635797144:{"name":"default-1635797144","paths":["PATH","stream-mode":"sample","encoding":"json"}]
password:
2021/11/01 21:05:48.285936 /build/config/targets.go:49: [config] targets: map[1.1.1.1:{"name":"1.1.1.1","address":"1.1.1.1:57400","username":"USERNAME","password":"PASSWORD",
"timeout":10000000000,"insecure":false,"tls-ca":"ca-cert.pem","tls-cert":"cert.pem","tls-key":"key.pem","skip-verify":false,"retry-timer":10000000000,"gzip":false,"token
":""}]
2021/11/01 21:05:48.286011 /build/config/inputs.go:49: [config] inputs: map[]
2021/11/01 21:05:48.286101 /build/config/subscriptions.go:44: [config] subscriptions: map[default-1635797144:{"name":"default-1635797144","paths":["PATH"],"mode":"s
tream","stream-mode":"sample","encoding":"json"} default-1635797148:{"name":"default-1635797148","paths":["PATH"],"mode":"stream","stream-mode":"sample","encoding":
"json"}]
2021/11/01 21:05:48.286210 /build/config/outputs.go:52: [config] outputs: map[default-stdout:map[file-type:stdout format: type:file]]
2021/11/01 21:05:48.286242 /build/config/processors.go:37: [config] processors: map[]
2021/11/01 21:05:48.286298 /build/app/subscribe.go:293: [gnmic] starting collector with config &{Name: Debug:true Format: TargetReceiveBuffer:0 RetryTimer:10s ClusterName:default-cluster LockRetryT
imer:5s}
2021/11/01 21:05:48.286382 /build/collector/collector.go:104: [gnmic] starting collector with cfg=&{Name: Debug:true Format: TargetReceiveBuffer:1000 RetryTimer:10s ClusterName:default-cluster Lock
RetryTimer:5s}
2021/11/01 21:05:48.286495 /build/collector/collector.go:182: [gnmic] adding target {"name":"1.1.1.1","address":"1.1.1.1:57400","username":"USERNAME","password":"PASSWORD","timeou
t":10000000000,"insecure":false,"tls-ca":"ca-cert.pem","tls-cert":"cert.pem","tls-key":"key.pem","skip-verify":false,"retry-timer":10000000000,"gzip":false,"token":""}
2021/11/01 21:05:48.286565 /build/collector/outputs.go:36: [gnmic] starting output type file
2021/11/01 21:05:48.287581 /build/outputs/file/file_output.go:174: [file_output] initialized file output: {"Cfg":{"FileName":"","FileType":"stdout","Format":"json","Multiline":true,"Indent":"  ","S
eparator":"\n","OverrideTimestamps":false,"AddTarget":"","TargetTemplate":"","EventProcessors":null,"ConcurrencyLimit":1000,"EnableMetrics":false,"Debug":false}}
2021/11/01 21:05:48.287877 /build/collector/collector.go:288: [gnmic] queuing target "1.1.1.1"
2021/11/01 21:05:48.287920 /build/collector/collector.go:290: [gnmic] subscribing to target: "1.1.1.1"
2021/11/01 21:05:48.288060 /build/collector/collector.go:561: [gnmic] starting target &{Config:{"name":"1.1.1.1","address":"1.1.1.1:57400","username":"USERNAME","password":"PASSWORD
ar","timeout":10000000000,"insecure":false,"tls-ca":"ca-cert.pem","tls-cert":"cert.pem","tls-key":"key.pem","skip-verify":false,"buffer-size":1000,"retry-timer":10000000
000,"gzip":false,"token":""} Subscriptions:map[default-1635797144:{"name":"default-1635797144","paths":["PATH"],"mode":"stream","stream-mode":"sample","encoding":"j
son"} default-1635797148:{"name":"default-1635797148","paths":["PATH"],"mode":"stream","stream-mode":"sample","encoding":"json"}] m:0xc0000dc078 Client:<nil> Subscr
ibeClients:map[] subscribeCancelFn:map[] pollChan:0xc00071c060 subscribeResponses:0xc00063a0c0 errors:0xc00071c0c0 stopped:false StopChan:0xc00071c120 Cfn:<nil> RootDesc:<nil>}
2021/11/01 21:05:48.288100 /build/collector/collector.go:573: [gnmic] starting target "1.1.1.1" listener
2021/11/01 21:05:48.289530 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] parsed scheme: ""
2021/11/01 21:05:48.289673 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] scheme "" not registered, fallback to default scheme
2021/11/01 21:05:48.289760 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] ccResolverWrapper: sending update to cc: {[{1.1.1.1:57400  <nil> 0 <nil>}] <nil> <nil
>}
2021/11/01 21:05:48.289803 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] ClientConn switching balancer to "pick_first"
2021/11/01 21:05:48.289848 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] Channel switches to new LB policy "pick_first"
2021/11/01 21:05:48.289940 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] Subchannel Connectivity change to CONNECTING
2021/11/01 21:05:48.290018 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] Subchannel picks a new address "1.1.1.1:57400" to connect
2021/11/01 21:05:48.290127 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] pickfirstBalancer: UpdateSubConnState: 0xc000206180, {CONNECTING <nil>}
2021/11/01 21:05:48.290193 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] Channel Connectivity change to CONNECTING
2021/11/01 21:05:48.313024 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] Subchannel Connectivity change to READY
2021/11/01 21:05:48.313085 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] pickfirstBalancer: UpdateSubConnState: 0xc000206180, {READY <nil>}
2021/11/01 21:05:48.313108 /go/pkg/mod/google.golang.org/grpc@v1.40.0/grpclog/logger.go:53: [gnmic] [core] Channel Connectivity change to READY
2021/11/01 21:05:48.313143 /build/collector/collector.go:475: [gnmic] target '1.1.1.1' gNMI client created
2021/11/01 21:05:48.316191 /build/collector/collector.go:478: [gnmic] sending gNMI SubscribeRequest: subscribe='subscribe:{subscription:{path:{elem:{name:"PREFIX"} elem:{name:"cpu
"} elem:{name:"cpu"}} mode:SAMPLE}}', mode='STREAM', encoding='JSON', to 1.1.1.1
2021/11/01 21:05:48.316322 /build/collector/collector.go:478: [gnmic] sending gNMI SubscribeRequest: subscribe='subscribe:{subscription:{path:{elem:{name:"PREFIX} elem:{name:"cpu
"} elem:{name:"cpu"}} mode:SAMPLE}}', mode='STREAM', encoding='JSON', to 1.1.1.1
```

```
gnmic version
version : 0.20.0
 commit : 2b61b68
   date : 2021-10-19T23:37:42Z
 gitURL : https://github.com/karimra/gnmic
   docs : https://gnmic.kmrd.dev
```

**Comment by karimra** [comment]
Sorry I didn't notice this issue, let me have a look at it.

**Comment by karimra** [comment]
it looks like the config picks up 2 subscriptions: 
```bash
2021/11/01 21:05:48.286101 /build/config/subscriptions.go:44: [config] subscriptions: map[default-1635797144:{"name":"default-1635797144","paths":["PATH"],"mode":"s
tream","stream-mode":"sample","encoding":"json"} default-1635797148:{"name":"default-1635797148","paths":["PATH"],"mode":"stream","stream-mode":"sample","encoding":
"json"}]
```

Are you running the subscribe command in prompt mode ?

**Comment by fredsod** [comment]
I noticed that before the 'password:'-prompt there is only one subscription(1635797144) in the debug log, and after there are two (1635797144 + 1635797148).

 i'm starting the subscription with the following command(without prompt mode):

`gnmic -a <IP> -u <username> sub -e json --tls-cert cert.pem --tls-key key.pem --tls-ca ca-cert.pem --path <path> --stream-mode sample -d`

**Comment by karimra** [comment]
Thanks, I see the same behaviour when password is not given in the command but via prompt.
I will try to find out what's happening.

**Comment by karimra** [comment]
Hi, 

Thanks again for reporting this.
I believe it is fixed in v0.20.3, can you check on your side ?

**Comment by fredsod** [comment]
Hi
thank you for the fix, it works great.
